From 95d934de91eaf5e932da48d7ad48af01a30988d0 Mon Sep 17 00:00:00 2001
From: sebastien baudouin <sebastien.baudouin@windriver.com>
Date: Mon, 27 Apr 2015 08:57:34 +0200
Subject: [PATCH 4/6] qml_example: Make Graphic working on Qemu machine

Those modification are in theory not needed but unfortunately without it the
graphic is not working.
---
 main.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/main.cpp b/main.cpp
index 9414896..a60213b 100644
--- a/main.cpp
+++ b/main.cpp
@@ -18,6 +18,12 @@
 
 #include <QApplication>
 #include <QQuickView>
+
+//For QEMU
+#include <QScreen>
+#include <QQuickWindow>
+#include <QQuickItem>
+
 #include <systemd/sd-journal.h>
 
 #include "main.h"
@@ -41,6 +47,13 @@ int main(int argc, char *argv[])
                      &myClass, SLOT(cppSlot(QString)));
 
     view.setProperty("IVI-Surface-ID", QML_EXAMPLE_SURFACE_ID);
+
+    //For QEMU only
+    QSize size = app.primaryScreen()->size();
+    view.showFullScreen();
+    view.rootObject()->setWidth(size.width());
+    view.rootObject()->setHeight(size.height());
+
     view.show();
 
     return app.exec();
-- 
1.9.1

